game-core/src/main/java/games/strategy/engine/framework/startup/mc/GameSelectorModel.java

```diff
@@ -59,21 +64,19 @@ public void load(final GameChooserEntry entry) {
     ClientSetting.flush();
   }
 
-  public void load(final File file, final Component ui) {
-    if (!file.exists()) {
-      error("Could not find file:" + file, ui);
-      return;
-    }
-    if (file.isDirectory()) {
-      error("Cannot load a directory:" + file, ui);
-      return;
-    }
+  /**
+   * Loads game data by parsing a given file.
+   */
+  public void load(final File file) {
+    Preconditions.checkArgument(file.exists() && file.isFile(),
+        "File should exist at: " + file.getAbsolutePath());
+
     final GameData newData;
     try {
       // if the file name is xml, load it as a new game
       if (file.getName().toLowerCase().endsWith("xml")) {
-        try (InputStream fis = new FileInputStream(file)) {
-          newData = GameParser.parse(file.getAbsolutePath(), fis);
+        try (final InputStream inputStream = new FileInputStream(file)) {
```

Comment:
Variables in try with resource blocks are implicitly final.
Therefore the final keyword is not necessary.
That's also the reason why everywhere in the codebase there's no final keyword at this position.

Reply:
Good point, now fixed & I found the setting to not warn about implicitly final variables and turned that off.